presto-main/src/main/java/com/facebook/presto/operator/scalar/ListLiteralCast.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.function.OperatorType;
+import com.facebook.presto.spi.function.ScalarOperator;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.type.ArrayType;
+import com.facebook.presto.type.ListLiteralType;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+
+public class ListLiteralCast
+{
+    private ListLiteralCast()
+    {
+    }
+
+    @ScalarOperator(OperatorType.CAST)
+    @SqlType(ListLiteralType.NAME)
+    public static List<Integer> castArrayToListLiteral(@SqlType("array(integer)") Block array)
```

Comment:
As per offline discussion, I'll leave it as is for now. I'll file an issue so we can come back to it later.